internal/controllers/machineset/machineset_controller.go

```diff
@@ -994,3 +1181,19 @@ func reconcileExternalTemplateReference(ctx context.Context, c client.Client, cl
 
 	return patchHelper.Patch(ctx, obj)
 }
+
+func skippedPreflightChecks(ms *clusterv1.MachineSet) sets.Set[clusterv1.MachineSetPreflightCheck] {
+	skippaed := sets.Set[clusterv1.MachineSetPreflightCheck]{}
+	if ms == nil {
+		return skippaed
+	}
+	skip := ms.Annotations[clusterv1.MachineSetSkipPreflightChecksAnnotation]
+	if skip == "" {
+		return skippaed
+	}
+	skippedList := strings.Split(skip, ",")
+	for i := range skippedList {
+		skippaed.Insert(clusterv1.MachineSetPreflightCheck(strings.TrimSpace(skippedList[i])))
```

Comment:
What happens if someone is using an invalid string?

Reply:
Effectively it will be ignored.
Example: If a user adds `foobar` in the annotation then it will be considered in the `skipped` set (the type conversion wont be a problem - any string can be converted to `clusterv1.MachineSetPreflightCheck`). Since there is not preflight check called `foobar` it will become no-op.

Reply:
Alright, thx for the info.

Can we add validation in the MS webhook to validate the list?

Reply:
Done. Also added the validation to MD webhook because users might choose to apply the annotation on an MD.